### ISSUE TITLE
GPU accleration for Density Matrix Simulator 

### ISSUE THREAD

**Issue by d-bharadwaj** [quantumlib/Cirq#6835]
**Is your feature request related to a use case or problem? Please describe.**

I want to enable GPU simulations on `cirq.DensityMatrixSimulator()`

**[optional] Describe alternatives/workarounds you've considered**

I have just been using CPU simulations as of now, which takes a lot of time for deep circuits.

**[optional] Additional context (e.g. screenshots)**

 I know qiskit has a way to run Density Matrix simulations on GPU and it is fast, so I was wondering if the feature is available yet on cirq.

**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by mhucka** [comment]
2024-12-11 discussed during Cirq Cynq: users could accomplish this in a couple of ways:
- perhaps use CuPy
- use qsim for GPU-based simulation

Concensus is that this is out of scope for implementation in Cirq itself.
